src/cdk/a11y/focus-trap.ts

```diff
@@ -337,7 +347,11 @@ export class FocusTrapDirective implements OnDestroy, AfterContentInit {
     this.focusTrap.destroy();
   }
 
-  ngAfterContentInit() {
-    this.focusTrap.attachAnchors();
+  ngDoCheck() {
+    // Since the element may not be attached to the DOM (or another element)
+    // immediately, keep trying until it is.
+    if (!this.focusTrap.hasAttached()) {
+      this.focusTrap.attachAnchors();
+    }
```

Comment:
Add a unit test for this?

Reply:
I had a hard time putting together a unit test that would fail as we expect it to. The only way I got the issue to happen at runtime was using a focus trap as the content of a select which has a complicated setup around it. I tried throwing a portal into the unit test, but it still worked.

Reply:
I ran into the issue with a setup like
```html
<ng-template cdkConnectedOverlay>
  <div cdkTrapFocus>...</div>
</ng-template>
```

Reply:
That doesn't seem to work either. During the test the focus trap still manages to find a parent node which allows it to attach (the ` <div id="root0">` in particular which AFAIK is the Karma root element), whereas in the demo app it doesn't, For reference, here's what the test component looks like:

```ts
@Component({
  template: `
    <button cdkOverlayOrigin #origin="cdkOverlayOrigin">I'm the origin</button>

    <ng-template cdkConnectedOverlay [open]="isOpen" [origin]="origin">
      <div cdkTrapFocus><button>I'm trapped!</button></div>
    </ng-template>
  `
})
class FocusTrapInOverlay {
  isOpen = false;
}
```

Reply:
The whole setup where it happened consistently was:
```html
<ng-template
    cdkConnectedOverlay
    [cdkConnectedOverlayOrigin]="trigger"
    [cdkConnectedOverlayOpen]="isOpen">

  <div cdkTrapFocus class="color-picker"
        role="dialog" aria-label="Color picker">
      <div role="listbox" tabindex="0" aria-label="Select a color"
        (keydown)="keydownHandler($event)">
        <div *ngFor="let color of colors; index as i"
            role="option"
            [style.background]="color.hex"
            [attr.aria-label]="color.name"
            [attr.aria-selected]="color.hex == selectedColor"
            (click)="selectedColor = color.hex; isOpen = false"></div>
      </div>
    </div>
</ng-template>
```
(ran into it while working on my angular mix demo)